serverless/serverless
# Added possibility to specify custom S3 key prefix instead of the stan…

## What did you implement:

Closes #5293

Ability to specify custom prefix under which artifacts are deployed to S3, which might be useful in scenarios where the default `serverless` prefix cannot be used (e.g. when using fine grained access control based on prefix).

## How did you implement it:

Added `getDeploymentPrefix` function to awsProvider.js that returns the custom prefix if defined (defaults to `serverless` if not defined) and replaced the default hardcoded prefix in various files with the call to `getDeploymentPrefix`.

## How can we verify it:

Add the `deploymentPrefix` key to the provider config as shown in the updated docs, run the deploy command, and check if resources are deployed under the specified prefix in S3

## Todos:


***Is this ready for review?:*** YES
***Is it a breaking change?:*** NO
